TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/EditTaskScreen/EditTaskScreenViewController.m

```diff
@@ -0,0 +1,359 @@
+       
+        NSDate *now = [NSDate dateWithTimeIntervalSinceNow:0];
+        NSUInteger nowMoment = [now timeIntervalSince1970];
+        NSUInteger delta = dateDueMoment - nowMoment;
+        
+        UNMutableNotificationContent *localNotification = [[UNMutableNotificationContent alloc] init];
+        localNotification.title = title;
+        localNotification.body = details;
+        localNotification.sound = [UNNotificationSound defaultSound];
+        
+        localNotification.badge = @([[UIApplication sharedApplication] applicationIconBadgeNumber] + 1);
+
+        UNTimeIntervalNotificationTrigger* trigger = [UNTimeIntervalNotificationTrigger triggerWithTimeInterval:delta repeats: NO];
+        
+        NSString *identifierLocalNotification = [NSString stringWithFormat:@"%ld", (long)[lid unsignedIntegerValue]];
+        
+        UNNotificationRequest *request = [UNNotificationRequest requestWithIdentifier:identifierLocalNotification
+                                                                              content:localNotification trigger:trigger];
+
+        UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
+        [center addNotificationRequest:request withCompletionHandler:^(NSError * _Nullable error) {
+            NSLog(@"Error: %@", [error localizedDescription]);
+        }];
+
+        nid = lid;
+        
+        [dictionaryTask setValue:nid forKey:@"nid"];
+        [singleton addTask:dictionaryTask taskId:lid];
+
+    
+    }
+    
+    [self.navigationController popViewControllerAnimated:YES];
+    
+}
+
+
+-(void) setDictionaryTask:(NSDictionary *)dictionaryTaskIn{
+    
+    dictionaryTask = [[NSMutableDictionary alloc] initWithDictionary:dictionaryTaskIn];
+    
+}
+
+
+-(void)showAlertMessage:(NSString *)title message:(NSString *)message{
+    
+    UIAlertController *alertController = [UIAlertController  alertControllerWithTitle:title  message:message preferredStyle:UIAlertControllerStyleAlert];
+    
+    UIAlertAction* ok = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil];
+    [alertController addAction:ok];
+    
+    [self presentViewController:alertController animated:YES completion:nil];
+    
+}
+
+
+
+- (void)textViewDidBeginEditing:(UITextView *)textView{
+
+    if (textView.tag == 2){
+
+        [self moveUp:shiftValue];
+
+    }
+    
+}
+
+
+- (void)textViewDidEndEditing:(UITextView *)textView{
+    
+
+    if (textView.tag == 2){
+        
+        [self moveDown:shiftValue];
+
+    }
+    
+}
+
+
+-(void)moveUp:(NSNumber *)moveUpValue{
+    
+    NSLog(@"moveUp");
+    
+    [UIView animateWithDuration:0.25 delay:0 options:UIViewAnimationOptionBeginFromCurrentState animations:^{
+        self.view.frame = CGRectMake(0, self.view.frame.origin.y - [moveUpValue floatValue], self.view.frame.size.width, self.view.frame.size.height);
+        
+    }completion:^(BOOL finished) {
+        if (finished) {
```

Comment:
Dangling code.

Reply:
+ retain cycle for `self` variable
+ hardcoded keyboard height in `shiftValue`. Here better to use keyboard observers